clients/admin-ui/src/features/common/nav/v2/nav-config.ts

```diff
@@ -75,14 +124,58 @@ export type NavGroup = {
   children: Array<NavGroupChild>;
 };
 
+/**
+ * If a group contains only routes that the user cannot access, return false.
+ * An empty list of scopes is a special case where any scope works.
+ */
+const navGroupInScope = (
+  group: NavConfigGroup,
+  userScopes: ScopeRegistry[]
+) => {
+  if (group.routes.filter((route) => route.scopes.length === 0).length === 0) {
+    const allScopesAcrossRoutes = group.routes.reduce((acc, route) => {
```

Comment:
Great use of iterators 👀  I don't use `reduce` that often

Reply:
honestly I avoid it when possible because I always have to look up how to use it... but for things like this it's really useful